lib/matplotlib/axes/_axes.py

```diff
@@ -6638,22 +6657,40 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
                     "sets and %d colors were provided" % (nx, len(color)))
                 raise ValueError(error_message)
 
-        # If bins are not specified either explicitly or via range,
-        # we need to figure out the range required for all datasets,
-        # and supply that to np.histogram.
-        if not binsgiven and not input_empty:
+        hist_kwargs = dict()
+
+        # if the bin_range is not given, compute without nan numpy
+        # does not do this for us when guessing the range (but will
+        # happily ignore nans when computing the histogram).
+        if bin_range is None:
```

Comment:
just `xmin = min((np.nanmin(xi) for xi in x if xi.size), default=np.inf)` and the same for xmax?  seems clearer